src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
+                size++;
+                return;
+            }
+            node = node.getNext();
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Node node = last;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        validateIndex(index);
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (index == i) {
+                return (T) node.getItem();
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
+        validateIndex(index);
+        final T replacedElement;
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (index == i) {
+                replacedElement = (T) node.getItem();
+                node.setItem(value);
+                return replacedElement;
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
+        validateIndex(index);
+        final T removedElement;
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (index == i) {
+                if (i == 0 && size == 1) {
+                    removedElement = (T) node.getItem();
+                    first = null;
+                } else if (i == 0) {
+                    removedElement = (T) node.getItem();
+                    node.getNext().setPrev(node.getPrev());
+                    first = node.getNext();
+                } else if (i == size - 1) {
+                    removedElement = (T) node.getItem();
+                    node.getPrev().setNext(node.getNext());
+                    last = node.getPrev();
+                } else {
+                    removedElement = (T) node.getItem();
+                    node.getPrev().setNext(node.getNext());
+                    node.getNext().setPrev(node.getPrev());
+                }
+                size--;
+                return removedElement;
+            }
+            node = node.getNext();
+        }
         return null;
     }
 
     @Override
     public boolean remove(T object) {
+        Node node = first;
+        for (int i = 0; i < size; i++) {
+            if (node.item == object || (node.item != null && node.item.equals(object))) {
+                if (i == 0 && size == 1) {
+                    first = null;
+                } else if (i == 0) {
+                    node.getNext().setPrev(node.getPrev());
+                    first = node.getNext();
+                } else if (i == size - 1) {
+                    node.getPrev().setNext(node.getNext());
+                    last = node.getPrev();
+                } else {
+                    node.getPrev().setNext(node.getNext());
+                    node.getNext().setPrev(node.getPrev());
+                }
+                size--;
```

Comment:
move it to ulink method [(see common mistakes)](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#create-a-separate-method-unlinknode-node)

Reply:
done